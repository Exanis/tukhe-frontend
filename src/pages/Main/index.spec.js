import React from 'react';
import { shallow } from 'enzyme';
import { styles, mapStateToProps, Main } from './index';

const props = {
    'setToken': jest.fn(),
    'updateWidgetData': jest.fn(),
    'addToWidgetData': jest.fn(),
    loadDashboards: jest.fn(),
    createDashboard: jest.fn(),
    selectDashboard: jest.fn(),
    loadAccounts: jest.fn(),
    deleteAccount: jest.fn(),
    createTwitterAccount: jest.fn(),
    createWidget: jest.fn(),
    updateDashboardLayout: jest.fn(),
    refreshWidgetWithoutAccount: jest.fn(),
    refreshWidgetWithAccount: jest.fn(),
    widgetActionWithAccount: jest.fn(),
    widgetActionWithoutAccount: jest.fn(),
    startWidgetProcess: jest.fn(),
    stopWidgetProcess: jest.fn(),
    dashboard: {
        currentDashboard: 'test',
        dashboards: [{
            uuid: 'test',
            layout: {},
            icon: 'home'
        },],
        widgets: [],
        data: {}
    },
    account: {
        accounts: []
    },
    classes: {}
};
const wrapper = shallow(<Main {...props} />);

it('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(props.loadDashboards).toHaveBeenCalled();
    expect(props.loadAccounts).toHaveBeenCalled();
    wrapper.setProps({
        account: {
            accounts: [],
            token: 'test'
        }
    });
    expect(wrapper).toMatchSnapshot();
});

it('Should manage popup correctly', () => {
    wrapper.instance().displayElementInPopup('test');
    expect(wrapper.state('popupDisplayElement')).toEqual('test');
    wrapper.instance().displayElementInPopup('test2', 'props');
    expect(wrapper.state('popupDisplayElement')).toEqual('test2');
    expect(wrapper.state('popupProps')).toEqual('props');
    wrapper.instance().closeElementPopup();
    expect(wrapper.state('popupDisplayElement')).toBeNull();
    expect(wrapper.state('popupProps')).toEqual({});
    wrapper.instance().updatePopupProps({
        'data1': 'test',
        'data2': 'other test'
    });
    wrapper.instance().updatePopupProps({
        'data1': 'something',
        'data3': 'else'
    });
    expect(wrapper.state('popupProps')).toEqual({
        'data1': 'something',
        'data2': 'other test',
        'data3': 'else'
    })
});

it('Should manage dashboard correctly', () => {
    wrapper.instance().openCreateDashboard();
    expect(wrapper.state('displayCreateDashboard')).toBeTruthy();
    wrapper.instance().createDashboard('test');
    expect(props.createDashboard).toHaveBeenCalledWith('test');
    expect(wrapper.state('displayCreateDashboard')).toBeFalsy();
    wrapper.instance().openCreateDashboard();
    wrapper.instance().onCloseCreateDashboard();
    expect(wrapper.state('displayCreateDashboard')).toBeFalsy();
    wrapper.instance().unlockDashboard();
    expect(wrapper.state('editable')).toBeTruthy();
    wrapper.instance().onLayoutChange({'test': 'test'});
    expect(wrapper.state('layout')).toEqual({'test': 'test'});
    wrapper.instance().onSaveDashboardLayout();
    expect(wrapper.state('editable')).toBeFalsy();
    expect(props.updateDashboardLayout).toHaveBeenCalledWith({'test': 'test'});
    wrapper.instance().onLayoutChange({'test': 'test2'});
    expect(wrapper.state('layout')).toEqual({'test': 'test'});
    wrapper.instance().refreshLayoutState({'test': 'new test'})();
    expect(wrapper.state('layout')).toEqual({'test': 'new test'});
});

it('Should manage widget correctly', () => {
    wrapper.instance().onAddWidget();
    expect(wrapper.state('displayCatalog')).toBeTruthy();
    wrapper.instance().onCloseCatalog();
    expect(wrapper.state('displayCatalog')).toBeFalsy();
    wrapper.instance().onAddWidget();
    wrapper.instance().createWidget('provider', 'widget');
    expect(wrapper.state('displayCatalog')).toBeFalsy();
    expect(wrapper.state('createWidgetTarget')).toEqual('provider_widget');
    expect(wrapper.state('displayCreateWidget')).toBeTruthy();
    wrapper.instance().onCreateWidget('test');
    expect(wrapper.state('displayCreateWidget')).toBeFalsy();
    expect(props.createWidget).toHaveBeenCalledWith('test');
    wrapper.instance().createWidget('provider', 'widget');
    wrapper.instance().onCloseNewWidget();
    expect(wrapper.state('displayCreateWidget')).toBeFalsy();
});

it('Should log out', () => {
    wrapper.instance().onLogout();
    expect(props.setToken).toHaveBeenCalledWith(null);
});

it('Should manage account', () => {
    wrapper.instance().onAccount();
    expect(wrapper.state('displayAccountList')).toBeTruthy();
    wrapper.instance().onCloseAccountList();
    expect(wrapper.state('displayAccountList')).toBeFalsy();
});

it('Should keep helper func coherent', () => {
    expect(styles()).toMatchSnapshot();
    expect(mapStateToProps({
        dashboard: 'dashboard',
        account: {
            accounts: [
                'test'
            ]
        }
    })).toMatchSnapshot();
});
