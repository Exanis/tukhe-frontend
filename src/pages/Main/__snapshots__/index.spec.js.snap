// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should keep helper func coherent 1`] = `
Object {
  "dashboard": Object {
    "height": "calc(100% - 65px)",
    "paddingTop": "65px",
  },
}
`;

exports[`Should keep helper func coherent 2`] = `
Object {
  "account": Object {
    "accounts": Array [
      "test",
    ],
  },
  "dashboard": "dashboard",
}
`;

exports[`Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    account={
      Object {
        "accounts": Array [],
      }
    }
    addToWidgetData={[MockFunction]}
    classes={Object {}}
    createDashboard={[MockFunction]}
    createTwitterAccount={[MockFunction]}
    createWidget={[MockFunction]}
    dashboard={
      Object {
        "currentDashboard": "test",
        "dashboards": Array [
          Object {
            "icon": "home",
            "layout": Object {},
            "uuid": "test",
          },
        ],
        "data": Object {},
        "widgets": Array [],
      }
    }
    deleteAccount={[MockFunction]}
    loadAccounts={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    loadDashboards={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    refreshWidgetWithAccount={[MockFunction]}
    refreshWidgetWithoutAccount={[MockFunction]}
    selectDashboard={[MockFunction]}
    setToken={[MockFunction]}
    startWidgetProcess={[MockFunction]}
    stopWidgetProcess={[MockFunction]}
    updateDashboardLayout={[MockFunction]}
    updateWidgetData={[MockFunction]}
    widgetActionWithAccount={[MockFunction]}
    widgetActionWithoutAccount={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Navbar
          createDashboard={[Function]}
          current="test"
          dashboards={
            Array [
              Object {
                "icon": "home",
                "layout": Object {},
                "uuid": "test",
              },
            ]
          }
          editable={false}
          onAccount={[Function]}
          onAddWidget={[Function]}
          onLogout={[Function]}
          onSaveLayout={[Function]}
          onUnlock={[Function]}
          selectDashboard={[MockFunction]}
        />,
        <InjectIntl(RawCreateDashboard)
          createDashboard={[Function]}
          onClose={[Function]}
          open={false}
        />,
        <InjectIntl(RawAccountsList)
          accounts={Array []}
          onClose={[Function]}
          onDeleteAccount={[MockFunction]}
          open={false}
          providers={
            Object {
              "twitter": [MockFunction],
            }
          }
        />,
        <InjectIntl(RawCatalog)
          onClose={[Function]}
          onSelectWidget={[Function]}
          open={false}
          widgets={
            Object {
              "misc": Array [
                "rss",
              ],
              "twitter": Array [
                "feed",
                "notifications",
              ],
            }
          }
        />,
        <InjectIntl(RawNewWidget)
          accounts={Array []}
          onClose={[Function]}
          onCreateWidget={[Function]}
          open={false}
          widget=""
        />,
        <WithStyles(RawDashboard)
          addToWidgetData={[MockFunction]}
          closeElementPopup={[Function]}
          data={Object {}}
          displayElementInPopup={[Function]}
          editable={false}
          layout={null}
          onLayoutChange={[Function]}
          refreshWithAccount={[MockFunction]}
          refreshWithoutAccount={[MockFunction]}
          startWidgetProcess={[MockFunction]}
          stopWidgetProcess={[MockFunction]}
          updatePopupProps={[Function]}
          updateWidgetData={[MockFunction]}
          widgetActionWithAccount={[MockFunction]}
          widgetActionWithoutAccount={[MockFunction]}
          widgets={Array []}
        />,
        <BasicPopup
          onClose={[Function]}
          open={false}
        />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "createDashboard": [Function],
          "current": "test",
          "dashboards": Array [
            Object {
              "icon": "home",
              "layout": Object {},
              "uuid": "test",
            },
          ],
          "editable": false,
          "onAccount": [Function],
          "onAddWidget": [Function],
          "onLogout": [Function],
          "onSaveLayout": [Function],
          "onUnlock": [Function],
          "selectDashboard": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "createDashboard": [Function],
          "onClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accounts": Array [],
          "onClose": [Function],
          "onDeleteAccount": [MockFunction],
          "open": false,
          "providers": Object {
            "twitter": [MockFunction],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClose": [Function],
          "onSelectWidget": [Function],
          "open": false,
          "widgets": Object {
            "misc": Array [
              "rss",
            ],
            "twitter": Array [
              "feed",
              "notifications",
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accounts": Array [],
          "onClose": [Function],
          "onCreateWidget": [Function],
          "open": false,
          "widget": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addToWidgetData": [MockFunction],
          "closeElementPopup": [Function],
          "data": Object {},
          "displayElementInPopup": [Function],
          "editable": false,
          "layout": null,
          "onLayoutChange": [Function],
          "refreshWithAccount": [MockFunction],
          "refreshWithoutAccount": [MockFunction],
          "startWidgetProcess": [MockFunction],
          "stopWidgetProcess": [MockFunction],
          "updatePopupProps": [Function],
          "updateWidgetData": [MockFunction],
          "widgetActionWithAccount": [MockFunction],
          "widgetActionWithoutAccount": [MockFunction],
          "widgets": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": null,
          "onClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Navbar
            createDashboard={[Function]}
            current="test"
            dashboards={
              Array [
                Object {
                  "icon": "home",
                  "layout": Object {},
                  "uuid": "test",
                },
              ]
            }
            editable={false}
            onAccount={[Function]}
            onAddWidget={[Function]}
            onLogout={[Function]}
            onSaveLayout={[Function]}
            onUnlock={[Function]}
            selectDashboard={[MockFunction]}
          />,
          <InjectIntl(RawCreateDashboard)
            createDashboard={[Function]}
            onClose={[Function]}
            open={false}
          />,
          <InjectIntl(RawAccountsList)
            accounts={Array []}
            onClose={[Function]}
            onDeleteAccount={[MockFunction]}
            open={false}
            providers={
              Object {
                "twitter": [MockFunction],
              }
            }
          />,
          <InjectIntl(RawCatalog)
            onClose={[Function]}
            onSelectWidget={[Function]}
            open={false}
            widgets={
              Object {
                "misc": Array [
                  "rss",
                ],
                "twitter": Array [
                  "feed",
                  "notifications",
                ],
              }
            }
          />,
          <InjectIntl(RawNewWidget)
            accounts={Array []}
            onClose={[Function]}
            onCreateWidget={[Function]}
            open={false}
            widget=""
          />,
          <WithStyles(RawDashboard)
            addToWidgetData={[MockFunction]}
            closeElementPopup={[Function]}
            data={Object {}}
            displayElementInPopup={[Function]}
            editable={false}
            layout={null}
            onLayoutChange={[Function]}
            refreshWithAccount={[MockFunction]}
            refreshWithoutAccount={[MockFunction]}
            startWidgetProcess={[MockFunction]}
            stopWidgetProcess={[MockFunction]}
            updatePopupProps={[Function]}
            updateWidgetData={[MockFunction]}
            widgetActionWithAccount={[MockFunction]}
            widgetActionWithoutAccount={[MockFunction]}
            widgets={Array []}
          />,
          <BasicPopup
            onClose={[Function]}
            open={false}
          />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "createDashboard": [Function],
            "current": "test",
            "dashboards": Array [
              Object {
                "icon": "home",
                "layout": Object {},
                "uuid": "test",
              },
            ],
            "editable": false,
            "onAccount": [Function],
            "onAddWidget": [Function],
            "onLogout": [Function],
            "onSaveLayout": [Function],
            "onUnlock": [Function],
            "selectDashboard": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "createDashboard": [Function],
            "onClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accounts": Array [],
            "onClose": [Function],
            "onDeleteAccount": [MockFunction],
            "open": false,
            "providers": Object {
              "twitter": [MockFunction],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClose": [Function],
            "onSelectWidget": [Function],
            "open": false,
            "widgets": Object {
              "misc": Array [
                "rss",
              ],
              "twitter": Array [
                "feed",
                "notifications",
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accounts": Array [],
            "onClose": [Function],
            "onCreateWidget": [Function],
            "open": false,
            "widget": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addToWidgetData": [MockFunction],
            "closeElementPopup": [Function],
            "data": Object {},
            "displayElementInPopup": [Function],
            "editable": false,
            "layout": null,
            "onLayoutChange": [Function],
            "refreshWithAccount": [MockFunction],
            "refreshWithoutAccount": [MockFunction],
            "startWidgetProcess": [MockFunction],
            "stopWidgetProcess": [MockFunction],
            "updatePopupProps": [Function],
            "updateWidgetData": [MockFunction],
            "widgetActionWithAccount": [MockFunction],
            "widgetActionWithoutAccount": [MockFunction],
            "widgets": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": null,
            "onClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Should render correctly 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    account={
      Object {
        "accounts": Array [],
        "token": "test",
      }
    }
    addToWidgetData={[MockFunction]}
    classes={Object {}}
    createDashboard={[MockFunction]}
    createTwitterAccount={[MockFunction]}
    createWidget={[MockFunction]}
    dashboard={
      Object {
        "currentDashboard": "test",
        "dashboards": Array [
          Object {
            "icon": "home",
            "layout": Object {},
            "uuid": "test",
          },
        ],
        "data": Object {},
        "widgets": Array [],
      }
    }
    deleteAccount={[MockFunction]}
    loadAccounts={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    loadDashboards={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    refreshWidgetWithAccount={[MockFunction]}
    refreshWidgetWithoutAccount={[MockFunction]}
    selectDashboard={[MockFunction]}
    setToken={[MockFunction]}
    startWidgetProcess={[MockFunction]}
    stopWidgetProcess={[MockFunction]}
    updateDashboardLayout={[MockFunction]}
    updateWidgetData={[MockFunction]}
    widgetActionWithAccount={[MockFunction]}
    widgetActionWithoutAccount={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Navbar
          createDashboard={[Function]}
          current="test"
          dashboards={
            Array [
              Object {
                "icon": "home",
                "layout": Object {},
                "uuid": "test",
              },
            ]
          }
          editable={false}
          onAccount={[Function]}
          onAddWidget={[Function]}
          onLogout={[Function]}
          onSaveLayout={[Function]}
          onUnlock={[Function]}
          selectDashboard={[MockFunction]}
        />,
        <InjectIntl(RawCreateDashboard)
          createDashboard={[Function]}
          onClose={[Function]}
          open={false}
        />,
        <InjectIntl(RawAccountsList)
          accounts={Array []}
          onClose={[Function]}
          onDeleteAccount={[MockFunction]}
          open={false}
          providers={
            Object {
              "twitter": [MockFunction],
            }
          }
        />,
        <InjectIntl(RawCatalog)
          onClose={[Function]}
          onSelectWidget={[Function]}
          open={false}
          widgets={
            Object {
              "misc": Array [
                "rss",
              ],
              "twitter": Array [
                "feed",
                "notifications",
              ],
            }
          }
        />,
        <InjectIntl(RawNewWidget)
          accounts={Array []}
          onClose={[Function]}
          onCreateWidget={[Function]}
          open={false}
          widget=""
        />,
        <WithStyles(RawDashboard)
          addToWidgetData={[MockFunction]}
          closeElementPopup={[Function]}
          data={Object {}}
          displayElementInPopup={[Function]}
          editable={false}
          layout={null}
          onLayoutChange={[Function]}
          refreshWithAccount={[MockFunction]}
          refreshWithoutAccount={[MockFunction]}
          startWidgetProcess={[MockFunction]}
          stopWidgetProcess={[MockFunction]}
          updatePopupProps={[Function]}
          updateWidgetData={[MockFunction]}
          widgetActionWithAccount={[MockFunction]}
          widgetActionWithoutAccount={[MockFunction]}
          widgets={Array []}
        />,
        <BasicPopup
          onClose={[Function]}
          open={false}
        />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "createDashboard": [Function],
          "current": "test",
          "dashboards": Array [
            Object {
              "icon": "home",
              "layout": Object {},
              "uuid": "test",
            },
          ],
          "editable": false,
          "onAccount": [Function],
          "onAddWidget": [Function],
          "onLogout": [Function],
          "onSaveLayout": [Function],
          "onUnlock": [Function],
          "selectDashboard": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "createDashboard": [Function],
          "onClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accounts": Array [],
          "onClose": [Function],
          "onDeleteAccount": [MockFunction],
          "open": false,
          "providers": Object {
            "twitter": [MockFunction],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClose": [Function],
          "onSelectWidget": [Function],
          "open": false,
          "widgets": Object {
            "misc": Array [
              "rss",
            ],
            "twitter": Array [
              "feed",
              "notifications",
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accounts": Array [],
          "onClose": [Function],
          "onCreateWidget": [Function],
          "open": false,
          "widget": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addToWidgetData": [MockFunction],
          "closeElementPopup": [Function],
          "data": Object {},
          "displayElementInPopup": [Function],
          "editable": false,
          "layout": null,
          "onLayoutChange": [Function],
          "refreshWithAccount": [MockFunction],
          "refreshWithoutAccount": [MockFunction],
          "startWidgetProcess": [MockFunction],
          "stopWidgetProcess": [MockFunction],
          "updatePopupProps": [Function],
          "updateWidgetData": [MockFunction],
          "widgetActionWithAccount": [MockFunction],
          "widgetActionWithoutAccount": [MockFunction],
          "widgets": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": null,
          "onClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Navbar
            createDashboard={[Function]}
            current="test"
            dashboards={
              Array [
                Object {
                  "icon": "home",
                  "layout": Object {},
                  "uuid": "test",
                },
              ]
            }
            editable={false}
            onAccount={[Function]}
            onAddWidget={[Function]}
            onLogout={[Function]}
            onSaveLayout={[Function]}
            onUnlock={[Function]}
            selectDashboard={[MockFunction]}
          />,
          <InjectIntl(RawCreateDashboard)
            createDashboard={[Function]}
            onClose={[Function]}
            open={false}
          />,
          <InjectIntl(RawAccountsList)
            accounts={Array []}
            onClose={[Function]}
            onDeleteAccount={[MockFunction]}
            open={false}
            providers={
              Object {
                "twitter": [MockFunction],
              }
            }
          />,
          <InjectIntl(RawCatalog)
            onClose={[Function]}
            onSelectWidget={[Function]}
            open={false}
            widgets={
              Object {
                "misc": Array [
                  "rss",
                ],
                "twitter": Array [
                  "feed",
                  "notifications",
                ],
              }
            }
          />,
          <InjectIntl(RawNewWidget)
            accounts={Array []}
            onClose={[Function]}
            onCreateWidget={[Function]}
            open={false}
            widget=""
          />,
          <WithStyles(RawDashboard)
            addToWidgetData={[MockFunction]}
            closeElementPopup={[Function]}
            data={Object {}}
            displayElementInPopup={[Function]}
            editable={false}
            layout={null}
            onLayoutChange={[Function]}
            refreshWithAccount={[MockFunction]}
            refreshWithoutAccount={[MockFunction]}
            startWidgetProcess={[MockFunction]}
            stopWidgetProcess={[MockFunction]}
            updatePopupProps={[Function]}
            updateWidgetData={[MockFunction]}
            widgetActionWithAccount={[MockFunction]}
            widgetActionWithoutAccount={[MockFunction]}
            widgets={Array []}
          />,
          <BasicPopup
            onClose={[Function]}
            open={false}
          />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "createDashboard": [Function],
            "current": "test",
            "dashboards": Array [
              Object {
                "icon": "home",
                "layout": Object {},
                "uuid": "test",
              },
            ],
            "editable": false,
            "onAccount": [Function],
            "onAddWidget": [Function],
            "onLogout": [Function],
            "onSaveLayout": [Function],
            "onUnlock": [Function],
            "selectDashboard": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "createDashboard": [Function],
            "onClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accounts": Array [],
            "onClose": [Function],
            "onDeleteAccount": [MockFunction],
            "open": false,
            "providers": Object {
              "twitter": [MockFunction],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClose": [Function],
            "onSelectWidget": [Function],
            "open": false,
            "widgets": Object {
              "misc": Array [
                "rss",
              ],
              "twitter": Array [
                "feed",
                "notifications",
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accounts": Array [],
            "onClose": [Function],
            "onCreateWidget": [Function],
            "open": false,
            "widget": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addToWidgetData": [MockFunction],
            "closeElementPopup": [Function],
            "data": Object {},
            "displayElementInPopup": [Function],
            "editable": false,
            "layout": null,
            "onLayoutChange": [Function],
            "refreshWithAccount": [MockFunction],
            "refreshWithoutAccount": [MockFunction],
            "startWidgetProcess": [MockFunction],
            "stopWidgetProcess": [MockFunction],
            "updatePopupProps": [Function],
            "updateWidgetData": [MockFunction],
            "widgetActionWithAccount": [MockFunction],
            "widgetActionWithoutAccount": [MockFunction],
            "widgets": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": null,
            "onClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
